{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "f79fc3c5",
   "metadata": {},
   "outputs": [],
   "source": [
    "from os import path, listdir, mkdir\n",
    "import warnings\n",
    "warnings.filterwarnings(\"ignore\", category=FutureWarning)\n",
    "import pandas as pd\n",
    "import matplotlib.pyplot as plt\n",
    "from matplotlib.lines import Line2D\n",
    "import seaborn as sns\n",
    "sns.set(rc={'grid.linewidth' : 1, 'figure.facecolor':'white'})\n",
    "sns.set(style = 'whitegrid')\n",
    "import numpy as np\n",
    "from scipy.stats import linregress\n",
    "from scipy.optimize import curve_fit\n",
    "from sklearn.metrics import r2_score"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c1c76cc6",
   "metadata": {},
   "source": [
    "Data processing pipeline:\n",
    "1. Download raw files from datasets PXD022836, PXD022868, PXD006442 (MS2) and PXD035055 (MS1)\n",
    "2. Convert raw files to mzML with $ \\textit{msConvert}$\n",
    "3. Perform peptide spectrum matching of LC-MS/MS data with search engines $ \\textit{IdentiPy}$, $\\textit{MSFragger}$, or $\\textit{X!Tandem}$. For LC-MS1 data $\\textit{ms1searchpy}$ can be used\n",
    "4. Perform search post-processing and calculation of NSAF values with Scavager\n",
    "5. Perform quantitative analysis with $\\textit{Diffacto}$ or with stat_test.ipynb from https://github.com/kazakova/Metrics\n",
    "6. Calculate metrics with metrics.ipynb from https://github.com/kazakova/Metrics\n",
    "7. Perform GO terms enrichment analysis with $\\textit{STRING}$ https://string-db.org/\n",
    "8. Calculate enrichment and GO score for processes from GO term enrichment analysis results (formulas can be found below)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c206c97f",
   "metadata": {},
   "source": [
    "$$ Enrichment = \\frac{proteome \\ size}{number \\ of \\ DE \\ genes} \\cdot \\frac{number \\ of \\ DE \\ genes \\ contributing \\ to \\ process}{number \\ of \\ genes \\ in \\ background} $$\n",
    "\n",
    "$$ GO \\ score = Enrichment \\cdot | log_{10} (fdr)| $$"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0ca3b62f",
   "metadata": {},
   "source": [
    "# Fig 1"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "112c08ae",
   "metadata": {},
   "source": [
    "Input files contain GO terms enrichment analysis results (step 7 from the pipeline)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "e37ecfec",
   "metadata": {},
   "outputs": [],
   "source": [
    "path_to_data = {'DBTRG&HOS IFNa' : '/home/kae-13-1/Metrics/Jan2022/18_01/MinDet/a172_dbtrg',\n",
    "                    'GBM2017 IFNa' : '/home/kae-13-1/Metrics/Jan2022/18_01/MinDet/glioblastoma_2017_infa',\n",
    "                    'GBM2017 IFNb' : '/home/kae-13-1/Metrics/Jan2022/18_01/MinDet/glioblastoma_2017_infb',\n",
    "                    'GBM2019 IFNa' : '/home/kae-13-1/Metrics/Jan2022/18_01/MinDet/glioblastoma_2019',\n",
    "                    'MRC5 IFNa': '/home/kae-13-1/Metrics/Jan2022/18_01/MinDet/time_infa',\n",
    "                    'MRC5 IFNg' : '/home/kae-13-1/Metrics/Jan2022/18_01/MinDet/time_infg'}\n",
    "\n",
    "dataname_to_plot = {'a172_dbtrg_DBTRG' :  'DBTRG IFNa-2b 1000 24h',\n",
    "                    'a172_dbtrg_A172' : 'HOS IFNa-2b 1000 24h',\n",
    "                    'glioblastoma_2019_AN' : 'AN IFNa-2b 100 24h',\n",
    "                    'glioblastoma_2019_6067' : 'GBM6067 IFNa-2b 100 24h',\n",
    "                    'glioblastoma_2019_3821' : 'GBM3821 IFNa-2b 100 24h',\n",
    "                    'glioblastoma_2019_5522' : 'GBM5522 IFNa-2b 100 24h',\n",
    "                    'glioblastoma_2019_4114' : 'GBM4114 IFNa-2b 100 24h',\n",
    "                    'glioblastoma_2017_infa_i' : 'GBM2017i IFNa-2b 100 24h',\n",
    "                    'glioblastoma_2017_infa_b' : 'GBM2017b IFNa-2b 100 24h',\n",
    "                    'glioblastoma_2017_infb_i' : 'GBM2017i IFNb-2b 1000 24h',\n",
    "                    'glioblastoma_2017_infb_b' : 'GBM2017b IFNb-2b 1000 24h',\n",
    "                    'time_infa_48' : 'MRC5 IFNa-2b 500 48h',\n",
    "                    'time_infa_24' : 'MRC5 IFNa-2b 500 24h',\n",
    "                    'time_infa_4' : 'MRC5 IFNa-2b 500 4h',\n",
    "                    'time_infg_48':'MRC5 IFNg 500 48h',\n",
    "                    'time_infg_24' : 'MRC5 IFNg 500 24h',\n",
    "                    'time_infg_4' : 'MRC5 IFNg 500 4h'}\n",
    "\n",
    "reg_type = '_GO'\n",
    "\n",
    "n_genes = pd.DataFrame()\n",
    "processes_set = set()\n",
    "\n",
    "ffolders = list(path_to_data.values())\n",
    "dirs = [f + '/GO' for f in ffolders]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "3628da0e",
   "metadata": {},
   "outputs": [],
   "source": [
    "for ffolder in ffolders:\n",
    "    for file in listdir(ffolder):\n",
    "        if (file.endswith(reg_type + '.csv')) and ('stat' in file):\n",
    "            df = pd.read_csv(path.join(ffolder, file))\n",
    "            dataname=  file.split(reg_type)[0] \n",
    "            n_genes.loc[dataname, 'n'] = df.shape[0]\n",
    "###################            \n",
    "for ffolder in dirs:\n",
    "    for file in listdir(ffolder):\n",
    "        if (file.endswith(reg_type + '.string.enrichment.tsv')) and ('stat' in file):\n",
    "            d = pd.read_csv(path.join(ffolder, file), sep = '\\t')\n",
    "            d = d[d['category'] == 'Process']\n",
    "            d['-log10(fdr)'] = d['fdr'].apply(lambda x: -np.log10(x))\n",
    "            \n",
    "            n_g = n_genes.loc[file.split(reg_type)[0], 'n']\n",
    "            \n",
    "            d['enrichment'] = d['number_of_genes'] * 20000/d['number_of_genes_in_background']/n_g\n",
    "            d['enrichment'] = d['enrichment'].apply(lambda x: np.log10(x))\n",
    "            \n",
    "            d['metric'] = d['-log10(fdr)'] * d['enrichment']\n",
    "            \n",
    "            d = d.sort_values(by = 'metric', ascending = False)\n",
    "            df = d.head(10)\n",
    "            tmp = list(df['description'].values)\n",
    "        \n",
    "            processes_set.update(tmp)\n",
    "##################\n",
    "processes_stat = pd.DataFrame(list(processes_set), columns = ['description'])\n",
    "for ffolder in dirs:\n",
    "    for file in listdir(ffolder):\n",
    "        if (file.endswith(reg_type + '.string.enrichment.tsv')) and ('stat' in file): \n",
    "            dataname = file.split('_stat')[0] + file.split('_stat')[1].split(reg_type)[0]   \n",
    "            d = pd.read_csv(path.join(ffolder, file), sep = '\\t')\n",
    "            d = d[d['category'] == 'Process']\n",
    "            d['-log10(fdr)'] = d['fdr'].apply(lambda x: -np.log10(x))\n",
    "            \n",
    "            n_g = n_genes.loc[file.split(reg_type)[0], 'n']\n",
    "                \n",
    "            d['enrichment'] = d['number_of_genes'] * 20000/d['number_of_genes_in_background']/n_g\n",
    "            d['enrichment'] = d['enrichment'].apply(lambda x: np.log10(x))\n",
    "            \n",
    "            d['metric'] = d['-log10(fdr)'] * d['enrichment']\n",
    "            \n",
    "            processes_stat = processes_stat.merge(right = d[['description','metric']], how = 'left',\n",
    "                                                               on = 'description')\n",
    "            processes_stat = processes_stat.rename(columns = {'metric' : dataname_to_plot[dataname]})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "7951cdaf",
   "metadata": {},
   "outputs": [],
   "source": [
    "colors = sns.color_palette('Set2', 6)\n",
    "legend_elements = [\n",
    "                   Line2D([0], [0], color = 'w', markerfacecolor = colors[0], marker = 's', markersize = 10,\n",
    "                          markeredgecolor = 'k',  markeredgewidth = .1, label = 'HOS IFNa 1000'),\n",
    "                   Line2D([0], [0], color = 'w', markerfacecolor = colors[2], marker = 's', markersize = 10,\n",
    "                          markeredgecolor = 'k',  markeredgewidth = .1, label = 'GBM2017b IFNa 100'),\n",
    "                   Line2D([0], [0], color = 'w', markerfacecolor = colors[4], marker = 's', markersize = 10,\n",
    "                          markeredgecolor = 'k',  markeredgewidth = .1, label = 'MRC5 IFNa 500'),\n",
    "                   Line2D([0], [0], color = 'w', markerfacecolor = colors[5], marker = 's', markersize = 10,\n",
    "                          markeredgecolor = 'k',  markeredgewidth = .1, label = 'MRC5 IFNg 500'),\n",
    "                   Line2D ([0], [0], color = 'w', label = 'Positive controls'),\n",
    "                   Line2D([0], [0], color = 'w', markerfacecolor = colors[0], marker = 'o', markersize = 10,\n",
    "                          markeredgecolor = 'k',  markeredgewidth = .1, label = 'DBTRG IFNa 1000'),\n",
    "                   Line2D([0], [0], color = 'w', markerfacecolor = colors[1], marker = 'o', markersize = 10,\n",
    "                          markeredgecolor = 'k',  markeredgewidth = .1, label = 'GBM2019 IFNa 100'),\n",
    "                   Line2D([0], [0], color = 'w', markerfacecolor = colors[2], marker = 'o', markersize = 10,\n",
    "                          markeredgecolor = 'k',  markeredgewidth = .1, label = 'GBM2017i IFNa 100'),\n",
    "                   Line2D([0], [0], color = 'w', markerfacecolor = colors[3], marker = 'o', markersize = 10,\n",
    "                          markeredgecolor = 'k',  markeredgewidth = .1, label = 'GBM2017i IFNb 1000'),\n",
    "                   Line2D ([0], [0], color = 'w', label = 'Negative controls'),\n",
    "                   Line2D([0], [0], color = 'w', markerfacecolor = colors[1], marker = 'X', markersize = 10,\n",
    "                          markeredgecolor = 'k',  markeredgewidth = .1, label = 'GBM4114 IFNa 100'),\n",
    "                   Line2D([0], [0], color = 'w', markerfacecolor = colors[3], marker = 'X', markersize = 10,\n",
    "                          markeredgecolor = 'k',  markeredgewidth = .1, label = 'GBM2017b IFNb 1000'),\n",
    "                 ]\n",
    "hue_order = ['DBTRG&HOS IFNa', 'GBM2019 IFNa', 'GBM2017 IFNa', 'GBM2017 IFNb', 'MRC5 IFNa', 'MRC5 IFNg']\n",
    "\n",
    "pos = ['GBM6067 IFNa-2b 100 24h', 'AN IFNa-2b 100 24h', 'GBM2017i IFNb-2b 1000 24h',\n",
    "      'GBM2017i IFNa-2b 100 24h', 'GBM5522 IFNa-2b 100 24h', 'GBM3821 IFNa-2b 100 24h',\n",
    "      'DBTRG IFNa-2b 1000 24h']\n",
    "\n",
    "neg = ['GBM2017b IFNb-2b 1000 24h', 'GBM4114 IFNa-2b 100 24h']\n",
    "\n",
    "markers = {\"pos\": \"o\", \n",
    "           \"neg\": \"X\",\n",
    "            'notcontrol' : 's'}\n",
    "\n",
    "data_to_dataset = {'GBM6067 IFNa-2b 100 24h': 'GBM2019 IFNa',\n",
    "                     'AN IFNa-2b 100 24h': 'GBM2019 IFNa',\n",
    "                     'GBM2017i IFNb-2b 1000 24h': 'GBM2017 IFNb',\n",
    "                     'GBM2017i IFNa-2b 100 24h': 'GBM2017 IFNa',\n",
    "                     'MRC5 IFNa-2b 500 48h': 'MRC5 IFNa',\n",
    "                     'HOS IFNa-2b 1000 24h': 'DBTRG&HOS IFNa',\n",
    "                     'GBM5522 IFNa-2b 100 24h': 'GBM2019 IFNa',\n",
    "                     'GBM3821 IFNa-2b 100 24h': 'GBM2019 IFNa',\n",
    "                     'DBTRG IFNa-2b 1000 24h': 'DBTRG&HOS IFNa',\n",
    "                     'GBM4114 IFNa-2b 100 24h': 'GBM2019 IFNa',\n",
    "                     'MRC5 IFNa-2b 500 24h': 'MRC5 IFNa',\n",
    "                     'MRC5 IFNg 500 48h': 'MRC5 IFNg',\n",
    "                     'MRC5 IFNa-2b 500 4h': 'MRC5 IFNa',\n",
    "                     'GBM2017b IFNa-2b 100 24h': 'GBM2017 IFNa',\n",
    "                     'MRC5 IFNg 500 24h': 'MRC5 IFNg',\n",
    "                     'MRC5 IFNg 500 4h': 'MRC5 IFNg',\n",
    "                     'GBM2017b IFNb-2b 1000 24h': 'GBM2017 IFNb'}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "8e7c7135",
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "#### Fig 1a\n",
    "sns.set_palette('Set2', 6)\n",
    "cm = 1/2.54\n",
    "processes = [processes_stat]\n",
    "\n",
    "names = ['MinDet \\n Static thresholds']\n",
    "\n",
    "plt_dir = '/home/kae-13-1/Metrics'\n",
    "\n",
    "for dataset, name in list(zip(processes, names)):\n",
    "    \n",
    "    tmp_df = pd.DataFrame(columns = ['description', 'value', 'data'])\n",
    "    dataset = dataset.set_index('description')\n",
    "    dataset = dataset.fillna(value = 0)\n",
    "    cols = dataset.columns\n",
    "    \n",
    "    dataset['maxscore'] = dataset.max(axis = 1)\n",
    "    dataset = dataset[dataset['maxscore'] >= 15]\n",
    "    dataset = dataset.drop(axis = 1, labels = ['maxscore'])\n",
    "    \n",
    "    for col in cols:\n",
    "        for i in dataset.index:\n",
    "            tmp = pd.DataFrame([[col, i, dataset.loc[i, col]]], columns = ['data', 'description', 'value'])\n",
    "            tmp_df = pd.concat([tmp_df, tmp], ignore_index = True)\n",
    "    tmp_df['description'] = tmp_df['description'].apply(lambda x: x.replace('Antigen processing and presentation', 'APP'))\n",
    "    \n",
    "    tmp_df = tmp_df.sort_values(by = 'value', ascending = False)\n",
    "    tmp_df['dataset'] = tmp_df.data.apply(lambda x: data_to_dataset[x])\n",
    "    tmp_df['controls'] = None\n",
    "    \n",
    "    for p in tmp_df.index:\n",
    "        if tmp_df.loc[p, 'data'] in pos:\n",
    "            tmp_df.loc[p, 'controls'] = 'pos'\n",
    "        elif tmp_df.loc[p, 'data'] in neg:\n",
    "            tmp_df.loc[p, 'controls'] = 'neg'\n",
    "        else:\n",
    "            tmp_df.loc[p, 'controls'] = 'notcontrol'\n",
    "    \n",
    "    y_len = len(tmp_df['description'].unique())\n",
    "    f, ax = plt.subplots(figsize = (13*cm, 10*cm))\n",
    "    sp = sns.scatterplot(x = 'value', y = 'description', hue = 'dataset',  hue_order = hue_order,\n",
    "                         edgecolor = 'black', linewidth = .1, style = 'controls', markers = markers,\n",
    "                         data = tmp_df, ax = ax, s = 15, alpha = 0.9)\n",
    "    ax.tick_params(axis = 'both', labelsize = 8)\n",
    "    ax.set_xlabel('Score', fontsize = 8)\n",
    "    ax.set_ylabel(None)\n",
    "    ax.set_title(name, fontsize = 8)\n",
    "    plt.legend(handles = legend_elements, fontsize = 6.5, markerscale = .5)\n",
    "    f.savefig(path.join(plt_dir, 'Fig1a.png'), dpi = 600, bbox_inches = 'tight')\n",
    "    f.savefig(path.join(plt_dir, 'Fig1a.svg'), dpi = 600, bbox_inches = 'tight')\n",
    "    plt.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "e1d24510",
   "metadata": {},
   "outputs": [],
   "source": [
    "#### Fig 1b\n",
    "sns.set_palette('Set2', 6)\n",
    "cols = [i for i in processes_stat.columns if 'MRC5' in i] + ['description']\n",
    "cm = 1/2.54\n",
    "\n",
    "processes = [processes_stat]\n",
    "names = ['MinDet \\n Static thresholds']\n",
    "plt_dir = '/home/kae-13-1/Metrics'\n",
    "hue_order = ['MRC5 IFNa-2b 500 48h',\n",
    "            'MRC5 IFNa-2b 500 24h',\n",
    "            'MRC5 IFNa-2b 500 4h',\n",
    "            'MRC5 IFNg 500 48h',\n",
    "            'MRC5 IFNg 500 24h',\n",
    "            'MRC5 IFNg 500 4h']\n",
    "\n",
    "for dataset, name in list(zip(processes, names)):\n",
    "    dataset = dataset[cols]\n",
    "    tmp_df = pd.DataFrame(columns = ['description', 'value', 'data'])\n",
    "    \n",
    "    dataset = dataset.set_index('description')\n",
    "    dataset = dataset.fillna(value = 0)\n",
    "    cols = dataset.columns\n",
    "    \n",
    "    dataset['maxscore'] = dataset.max(axis = 1)\n",
    "    dataset = dataset[dataset['maxscore'] >= 15] \n",
    "    dataset = dataset.drop(axis = 1, labels = ['maxscore']) \n",
    "    \n",
    "    for col in cols:\n",
    "        for i in dataset.index:\n",
    "            tmp = pd.DataFrame([[col, i, dataset.loc[i, col]]], columns = ['data', 'description', 'value'])\n",
    "            tmp_df = pd.concat([tmp_df, tmp], ignore_index = True)\n",
    "    tmp_df['description'] = tmp_df['description'].apply(lambda x: x.replace('Antigen processing and presentation', 'APP'))\n",
    "    \n",
    "    tmp_df = tmp_df.sort_values(by = 'value', ascending = False)\n",
    "    \n",
    "    y_len = len(tmp_df['description'].unique())\n",
    "    f, ax = plt.subplots(figsize = (13*cm, 4*cm))\n",
    "    sp = sns.scatterplot(x = 'value', y = 'description', hue = 'data', hue_order = hue_order,\n",
    "                         edgecolor = 'black', linewidth = .1, \n",
    "                         data = tmp_df, ax = ax, s = 15)\n",
    "    ax.tick_params(axis = 'both', labelsize = 8)\n",
    "\n",
    "    ax.set_xlabel('Score', fontsize = 8)\n",
    "    ax.set_ylabel(None)\n",
    "    ax.set_title(name, fontsize = 8)\n",
    "    plt.legend(fontsize = 6.5, markerscale = .5)\n",
    "    f.savefig(path.join(plt_dir, 'Fig1b.png'), dpi = 600, bbox_inches = 'tight')\n",
    "    f.savefig(path.join(plt_dir, 'Fig1b.svg'), dpi = 600, bbox_inches = 'tight')\n",
    "    plt.close()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "041532db",
   "metadata": {},
   "source": [
    "# Fig S1"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "93257250",
   "metadata": {},
   "source": [
    "Input files contain calculated metrics (step 6 from the pipeline)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "2e5f2950",
   "metadata": {},
   "outputs": [],
   "source": [
    "colors  = sns.color_palette('rainbow', 6)\n",
    "markers = {\"UP\": \"o\", \"DOWN\": \"X\"}\n",
    "hue_order = ['DBTRG&HOS IFNa',\n",
    "             'GBM2017 IFNa',\n",
    "             'GBM2017 IFNb',\n",
    "             'GBM2019 IFNa',\n",
    "             'MRC5 IFNa',\n",
    "             'MRC5 IFNg']\n",
    "\n",
    "metrics_data_dir = '/home/kae-13-1/Metrics/Mar2022/16_03'\n",
    "metrics_kNN_dyn = pd.DataFrame()\n",
    "metrics_kNN_semi_dyn = pd.DataFrame()\n",
    "metrics_kNN_stat = pd.DataFrame()\n",
    "metrics_MinDet_dyn = pd.DataFrame()\n",
    "metrics_MinDet_semi_dyn = pd.DataFrame()\n",
    "metrics_MinDet_stat = pd.DataFrame()\n",
    "\n",
    "for file in listdir(metrics_data_dir):\n",
    "    if file.startswith('metrics'):\n",
    "        d = pd.read_csv(path.join(metrics_data_dir, file), delimiter = '\\t', index_col = 0)\n",
    "        if 'knn' in file:\n",
    "            if '_dyn' in file:\n",
    "                metrics_kNN_dyn = pd.concat([metrics_kNN_dyn, d], axis = 1)\n",
    "            elif '_semi' in file:\n",
    "                metrics_kNN_semi_dyn = pd.concat([metrics_kNN_semi_dyn, d], axis = 1)\n",
    "            if '_stat' in file:\n",
    "                metrics_kNN_stat = pd.concat([metrics_kNN_stat, d], axis = 1)\n",
    "        elif 'mindet' in file:\n",
    "            if '_dyn' in file:\n",
    "                metrics_MinDet_dyn = pd.concat([metrics_MinDet_dyn, d], axis = 1)\n",
    "            elif '_semi' in file:\n",
    "                metrics_MinDet_semi_dyn = pd.concat([metrics_MinDet_semi_dyn, d], axis = 1)\n",
    "            if '_stat' in file:\n",
    "                metrics_MinDet_stat = pd.concat([metrics_MinDet_stat, d], axis = 1)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "1dbdd22d",
   "metadata": {},
   "outputs": [],
   "source": [
    "tick_colors = {'MRC5 IFNg 500 48h' : 'black', \n",
    "               'MRC5 IFNg 500 24h' : 'black', \n",
    "               'DBTRG IFNa-2b 1000 24h' : colors[5],\n",
    "               'HOS IFNa-2b 1000 24h' : 'black', \n",
    "               'AN IFNa-2b 100 24h' : colors[5],\n",
    "               'GBM5522 IFNa-2b 100 24h' : colors[5], \n",
    "               'GBM6067 IFNa-2b 100 24h' : colors[5],\n",
    "               'GBM2017i IFNa-2b 100 24h' : colors[5], \n",
    "               'MRC5 IFNg 500 4h': 'black',\n",
    "               'MRC5 IFNa-2b 500 48h' : 'black', \n",
    "               'MRC5 IFNa-2b 500 24h' : 'black',\n",
    "               'GBM2017i IFNb-2b 1000 24h' : colors[5], \n",
    "               'MRC5 IFNa-2b 500 4h' : 'black',\n",
    "               'GBM3821 IFNa-2b 100 24h' : colors[5], \n",
    "               'GBM2017b IFNa-2b 100 24h' : 'black',\n",
    "               'GBM4114 IFNa-2b 100 24h' : colors[0], \n",
    "               'GBM2017b IFNb-2b 1000 24h' : colors[0]}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "c91280bd",
   "metadata": {},
   "outputs": [],
   "source": [
    "sns.set_style('whitegrid', {'grid.linewidth' : 1})\n",
    "cm = 1/2.54\n",
    "\n",
    "metrics_MinDet = [metrics_MinDet_stat, metrics_MinDet_semi_dyn, metrics_MinDet_dyn]\n",
    "metrics_kNN = [metrics_kNN_stat, metrics_kNN_semi_dyn, metrics_kNN_dyn]\n",
    "metrics_dfs = [metrics_MinDet, metrics_kNN]\n",
    "\n",
    "metrics = ['E_dist', 'E-dist_mod', 'p1', 'log10(p2)']\n",
    "names = ['Euclidean distance', 'Modified euclidean distance', \n",
    "        '$ \\pi_{1} = \\sum_{i = 1}^{n} | \\  log_{2}FC_{i} \\cdot log_{10}FDR_{i} \\ | $', \n",
    "         '$ \\pi_{2} = log_{10} ( \\prod_{i = 1}^{n} \\ | \\  log_{2}FC_{i} \\cdot log_{10}FDR_{i} \\ | ) $']\n",
    "\n",
    "plt_order = [['a', 'e'], \n",
    "             ['b', 'f'],\n",
    "             ['c', 'g'],\n",
    "             ['d', 'h']]\n",
    "\n",
    "thr = ['Static thresholds', 'Semi-dynamic thresholds', 'Dynamic thresholds']\n",
    "\n",
    "plt_dir = '/home/kae-13-1/Metrics'\n",
    "i = 0\n",
    "for (m, txt) in zip(metrics, plt_order):\n",
    "    for metrics in metrics_dfs:\n",
    "        y_i = 0\n",
    "        fig, axes = plt.subplots(3, 1, figsize = (14*cm, 21*cm))\n",
    "        for df in metrics:\n",
    "            df = df.sort_values(by = m, ascending = False)\n",
    "            df = df.loc[:,~df.columns.duplicated()]\n",
    "            labels = df['dataname'].values\n",
    "            ax = axes[y_i]\n",
    "            f = sns.scatterplot(x = m, y = 'dataname', hue = 'Dataset', style = 'reg. type', markers = markers,\n",
    "                    edgecolor = 'black', linewidth = .1,\n",
    "                    hue_order = hue_order,\n",
    "                    data = df, marker = 'o',  ax = ax, s = 15)\n",
    "\n",
    "            ax.tick_params(axis = 'both', labelsize = 6)\n",
    "            ax.set_ylabel(None)\n",
    "            ax.set_xlabel(None)\n",
    "            ax.set_title(thr[y_i], fontsize = 6)\n",
    "            ax.text(-0.1, 1.1, txt[i%2] + str(y_i + 1), \n",
    "                    horizontalalignment = 'left', \n",
    "                    verticalalignment = 'top', \n",
    "                   transform  = ax.transAxes, fontsize = 6) \n",
    "            f.legend(fontsize = 6, markerscale =  .5)\n",
    "\n",
    "\n",
    "            y_i+= 1\n",
    "\n",
    "        fig.suptitle(names[i//2], fontsize = 7)\n",
    "        fig.supxlabel('Score', fontsize = 7)\n",
    "        plt.subplots_adjust(top = .92, hspace =  .3, bottom = .05)\n",
    "        \n",
    "        plt_name = 'FigS1_{}.png'.format(txt[i%2])\n",
    "        plt.draw()\n",
    "        for ax in axes:\n",
    "            [t.set_color(tick_colors[str(t).split(\"'\")[1]]) for t in ax.get_yticklabels()]\n",
    "        \n",
    "        plt.savefig(path.join(plt_dir, plt_name), bbox_inches = 'tight', dpi = 600)\n",
    "        plt.close()\n",
    "        i+=1\n",
    "    "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f64985d7",
   "metadata": {},
   "source": [
    "# Fig 3"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a2723b2c",
   "metadata": {},
   "source": [
    "Input files contain processes with calculated Enrichment and GO score (step 8 from the pipeline). Processes identified for all comparisons are listed in one dataframe (example can be found in the output)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "49a32051",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['Interferon-gamma-mediated signaling pathway', 'Type i interferon signaling pathway', 'Antigen processing and presentation of endogenous peptide antigen via mhc class i']\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>MRC5 IFNg 500 4h</th>\n",
       "      <th>MRC5 IFNa-2b 500 4h</th>\n",
       "      <th>MRC5 IFNg 500 24h</th>\n",
       "      <th>MRC5 IFNa-2b 500 48h</th>\n",
       "      <th>MRC5 IFNg 500 48h</th>\n",
       "      <th>MRC5 IFNa-2b 500 24h</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>description</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Cellular localization</th>\n",
       "      <td>0</td>\n",
       "      <td>14.455424</td>\n",
       "      <td>19.802825</td>\n",
       "      <td>5.768828</td>\n",
       "      <td>6.322638</td>\n",
       "      <td>27.706030</td>\n",
       "      <td>3.892063</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Regulated exocytosis</th>\n",
       "      <td>1</td>\n",
       "      <td>12.257196</td>\n",
       "      <td>10.682717</td>\n",
       "      <td>4.625328</td>\n",
       "      <td>3.105468</td>\n",
       "      <td>18.470909</td>\n",
       "      <td>3.348091</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Establishment of protein localization</th>\n",
       "      <td>2</td>\n",
       "      <td>6.807286</td>\n",
       "      <td>17.806405</td>\n",
       "      <td>2.837160</td>\n",
       "      <td>6.441568</td>\n",
       "      <td>13.864886</td>\n",
       "      <td>1.446335</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Generation of precursor metabolites and energy</th>\n",
       "      <td>3</td>\n",
       "      <td>2.839817</td>\n",
       "      <td>2.033889</td>\n",
       "      <td>12.562045</td>\n",
       "      <td>4.172144</td>\n",
       "      <td>17.473150</td>\n",
       "      <td>4.616226</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Cellular protein localization</th>\n",
       "      <td>4</td>\n",
       "      <td>8.066565</td>\n",
       "      <td>16.077823</td>\n",
       "      <td>2.246288</td>\n",
       "      <td>6.096045</td>\n",
       "      <td>16.541773</td>\n",
       "      <td>1.482911</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                Unnamed: 0  MRC5 IFNg 500 4h  \\\n",
       "description                                                                    \n",
       "Cellular localization                                    0         14.455424   \n",
       "Regulated exocytosis                                     1         12.257196   \n",
       "Establishment of protein localization                    2          6.807286   \n",
       "Generation of precursor metabolites and energy           3          2.839817   \n",
       "Cellular protein localization                            4          8.066565   \n",
       "\n",
       "                                                MRC5 IFNa-2b 500 4h  \\\n",
       "description                                                           \n",
       "Cellular localization                                     19.802825   \n",
       "Regulated exocytosis                                      10.682717   \n",
       "Establishment of protein localization                     17.806405   \n",
       "Generation of precursor metabolites and energy             2.033889   \n",
       "Cellular protein localization                             16.077823   \n",
       "\n",
       "                                                MRC5 IFNg 500 24h  \\\n",
       "description                                                         \n",
       "Cellular localization                                    5.768828   \n",
       "Regulated exocytosis                                     4.625328   \n",
       "Establishment of protein localization                    2.837160   \n",
       "Generation of precursor metabolites and energy          12.562045   \n",
       "Cellular protein localization                            2.246288   \n",
       "\n",
       "                                                MRC5 IFNa-2b 500 48h  \\\n",
       "description                                                            \n",
       "Cellular localization                                       6.322638   \n",
       "Regulated exocytosis                                        3.105468   \n",
       "Establishment of protein localization                       6.441568   \n",
       "Generation of precursor metabolites and energy              4.172144   \n",
       "Cellular protein localization                               6.096045   \n",
       "\n",
       "                                                MRC5 IFNg 500 48h  \\\n",
       "description                                                         \n",
       "Cellular localization                                   27.706030   \n",
       "Regulated exocytosis                                    18.470909   \n",
       "Establishment of protein localization                   13.864886   \n",
       "Generation of precursor metabolites and energy          17.473150   \n",
       "Cellular protein localization                           16.541773   \n",
       "\n",
       "                                                MRC5 IFNa-2b 500 24h  \n",
       "description                                                           \n",
       "Cellular localization                                       3.892063  \n",
       "Regulated exocytosis                                        3.348091  \n",
       "Establishment of protein localization                       1.446335  \n",
       "Generation of precursor metabolites and energy              4.616226  \n",
       "Cellular protein localization                               1.482911  "
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Reading data\n",
    "\n",
    "df4 = pd.read_csv('/home/kae-13-1/Metrics/time_inf_raw/identipy_scavager/metrics_res/processes_UP_identipy_diffacto_dyn.tsv', sep = '\\t', index_col = 'description')\n",
    "df5 = pd.read_csv('/home/kae-13-1/Metrics/time_inf_raw/identipy_scavager/metrics_res/processes_UP_identipy_diffacto_semi_dyn.tsv', sep = '\\t', index_col = 'description')\n",
    "df6 = pd.read_csv('/home/kae-13-1/Metrics/time_inf_raw/identipy_scavager/metrics_res/processes_UP_identipy_diffacto_stat.tsv', sep = '\\t', index_col = 'description')\n",
    "\n",
    "df7 = pd.read_csv('/home/kae-13-1/Metrics/time_inf_raw/frager_scavager/diffacto/metrics_res/processes_UP_msfragger_diffacto_dyn.tsv', sep = '\\t', index_col = 'description')\n",
    "df8 = pd.read_csv('/home/kae-13-1/Metrics/time_inf_raw/frager_scavager/diffacto/metrics_res/processes_UP_msfragger_diffacto_semi_dyn.tsv', sep = '\\t', index_col = 'description')\n",
    "df9 = pd.read_csv('/home/kae-13-1/Metrics/time_inf_raw/frager_scavager/diffacto/metrics_res/processes_UP_msfragger_diffacto_stat.tsv', sep = '\\t', index_col = 'description')\n",
    "\n",
    "df10 = pd.read_csv('/home/kae-13-1/Metrics/time_inf_raw/tandem_scavager/diffacto/metrics_res/processes_UP_xtandem_diffacto_dyn.tsv', sep = '\\t', index_col = 'description')\n",
    "df11 = pd.read_csv('/home/kae-13-1/Metrics/time_inf_raw/tandem_scavager/diffacto/metrics_res/processes_UP_xtandem_diffacto_semi_dyn.tsv', sep = '\\t', index_col = 'description')\n",
    "df12 = pd.read_csv('/home/kae-13-1/Metrics/time_inf_raw/tandem_scavager/diffacto/metrics_res/processes_UP_xtandem_diffacto_stat.tsv', sep = '\\t', index_col = 'description')\n",
    "\n",
    "\n",
    "#Filtering\n",
    "columns_order = [\n",
    "    'MRC5 IFNg 500 48h', 'MRC5 IFNg 500 24h', 'MRC5 IFNg 500 4h',\n",
    "    'MRC5 IFNa-2b 500 48h', 'MRC5 IFNa-2b 500 24h', 'MRC5 IFNa-2b 500 4h'\n",
    "]\n",
    "\n",
    "#Processes intersection\n",
    "pr = set.intersection( \n",
    "                      set(df4.index), set(df5.index), set(df6.index),\n",
    "                      set(df7.index), set(df8.index), set(df9.index),\n",
    "                      set(df10.index), set(df11.index), set(df12.index)\n",
    "                     )\n",
    "print(list(pr))\n",
    "\n",
    "#Shaping data\n",
    "cols =  ['D', 'SD', 'S']\n",
    "\n",
    "df_1 = pd.DataFrame(columns = cols)\n",
    "df_2 = pd.DataFrame(columns = cols)\n",
    "df_3 = pd.DataFrame(columns = cols)\n",
    "\n",
    "dfs = [ \n",
    "       df4, df5, df6,\n",
    "      df7, df8, df9,\n",
    "      df10, df11, df12]\n",
    "\n",
    "df_s = [df_1, df_2, df_3]\n",
    "\n",
    "for i in range(9):\n",
    "    dfi = dfs[i]\n",
    "    col_list = [c for c in columns_order if c in dfi.columns]\n",
    "    dfi = dfi[col_list]\n",
    "    dfi = dfi.loc['Type i interferon signaling pathway', :]\n",
    "    x = i%3\n",
    "    y = i//3\n",
    "    df_s[y][cols[x]] = dfi\n",
    "\n",
    "#Plot \n",
    "colors = sns.color_palette('Set2', 3)\n",
    "cm = 1/2.54\n",
    "f, axes = plt.subplots(1, 3,  figsize = (16*cm, 8*cm))\n",
    "names = ['IdentiPy\\ndiffacto',\n",
    "         'MSFragger\\ndiffacto',\n",
    "         'X!Tandem\\ndiffacto']\n",
    "\n",
    "for df_i, name, ax_i in list(zip(df_s, names, axes)):\n",
    "    \n",
    "    df_i['D'].plot(kind = 'bar', alpha = 0.9, position = 0, width = 0.3, color = colors[0], ax = ax_i)\n",
    "    df_i['SD'].plot(kind = 'bar', alpha = 0.9, position = 1, width = 0.3,  color = colors[1], ax = ax_i)\n",
    "    df_i['S'].plot(kind = 'bar', alpha = 0.9,  position = 2, width = 0.3, color = colors[2], ax = ax_i)\n",
    "    \n",
    "    ax_i.tick_params(axis = 'x', labelsize = 7)\n",
    "    ax_i.set_xticks(ticks = [0, 1, 2, 3, 4, 5],\n",
    "                    labels = ['IFNg 48h', 'IFNg 24h', 'IFNg 4h', \n",
    "                              'IFNa-2b 48h', 'IFNa-2b 24h', 'IFNa-2b 4h'])\n",
    "    ax_i.tick_params(axis = 'y', labelsize = 7)\n",
    "    ax_i.set_ylim(0, 45)\n",
    "    ax_i.set_xlim(-0.75, 5.5)\n",
    "    ax_i.set_title(name, fontsize = 8)\n",
    "      \n",
    "    if name == 'IdentiPy\\ndiffacto':\n",
    "        ax_i.set_ylabel('Type i interferon signaling pathway', fontsize = 7)\n",
    "        ax_i.legend(fontsize = 7, labels = ['dynamic', 'semi-dynamic', 'static'])\n",
    "    else:\n",
    "        ax_i.set_yticklabels([])\n",
    "\n",
    "plt.subplots_adjust(wspace = .05)\n",
    "f.savefig('/home/kae-13-1/Metrics/Fig3a.png', dpi = 600, bbox_inches = 'tight')    \n",
    "f.savefig('/home/kae-13-1/Metrics/Fig3a.svg', dpi = 600, bbox_inches = 'tight')\n",
    "\n",
    "plt.close()    \n",
    "\n",
    "df6.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "c42d7986",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['Innate immune response', 'Negative regulation of viral genome replication', 'Negative regulation of viral process', 'Interferon-gamma-mediated signaling pathway', 'Type i interferon signaling pathway', 'Antigen processing and presentation of endogenous peptide antigen via mhc class i', 'Response to type i interferon', 'Response to interferon-alpha', 'Defense response to virus']\n"
     ]
    }
   ],
   "source": [
    "##Fig3b\n",
    "#Reading data\n",
    "\n",
    "#X!Tandem + NSAF\n",
    "df1 = pd.read_csv('/home/kae-13-1/processes_kNN_dyn.tsv', sep = '\\t', index_col = 'description')\n",
    "df2 = pd.read_csv('/home/kae-13-1/processes_kNN_stat.tsv', sep = '\\t', index_col = 'description')\n",
    "df3 = pd.read_csv('/home/kae-13-1/processes_MinDet_dyn.tsv', sep = '\\t', index_col = 'description')\n",
    "df4 = pd.read_csv('/home/kae-13-1/processes_MinDet_stat.tsv', sep = '\\t', index_col = 'description')\n",
    "\n",
    "#Identipy+NSAF\n",
    "df5 = pd.read_csv('/home/kae-13-1/Metrics/Mar2022/16_03/processes_UP_kNN_dyn.tsv', sep = '\\t', index_col = 'description')\n",
    "df6 = pd.read_csv('/home/kae-13-1/Metrics/Mar2022/16_03/processes_UP_kNN_stat.tsv', sep = '\\t', index_col = 'description')\n",
    "df7 = pd.read_csv('/home/kae-13-1/Metrics/Mar2022/16_03/processes_UP_MinDet_dyn.tsv', sep = '\\t', index_col = 'description')\n",
    "df8 = pd.read_csv('/home/kae-13-1/Metrics/Mar2022/16_03/processes_UP_MinDet_stat.tsv', sep = '\\t', index_col = 'description')\n",
    "\n",
    "\n",
    "#Frager+NSAF\n",
    "df9 = pd.read_csv('/home/kae-13-1/Metrics/time_inf_raw/frager_scavager/processes_UP_fragger_nsaf_knn_dyn.tsv', sep = '\\t', index_col = 'description')\n",
    "df10 = pd.read_csv('/home/kae-13-1/Metrics/time_inf_raw/frager_scavager/processes_UP_fragger_nsaf_knn_stat.tsv', sep = '\\t', index_col = 'description')\n",
    "df11 = pd.read_csv('/home/kae-13-1/Metrics/time_inf_raw/frager_scavager/processes_UP_fragger_nsaf_mindet_dyn.tsv', sep = '\\t', index_col = 'description')\n",
    "df12 = pd.read_csv('/home/kae-13-1/Metrics/time_inf_raw/frager_scavager/processes_UP_fragger_nsaf_mindet_stat.tsv', sep = '\\t', index_col = 'description')\n",
    "\n",
    "\n",
    "columns_order_3 = [ 'MRC5 IFNg 500 48h', 'MRC5 IFNg 500 24h', 'MRC5 IFNg 500 4h',\n",
    "                    'MRC5 IFNa-2b 500 48h', 'MRC5 IFNa-2b 500 24h', 'MRC5 IFNa-2b 500 4h']\n",
    "\n",
    "#Processes intersection\n",
    "pr = set.intersection(set(df1.index), set(df2.index), set(df3.index), set(df4.index),\n",
    "                      set(df5.index), set(df6.index), set(df7.index), set(df8.index),\n",
    "                     set(df9.index), set(df10.index), set(df11.index), set(df12.index),)\n",
    "print(list(pr))\n",
    "\n",
    "#Shaping data\n",
    "dfs = [\n",
    "        df5, df6, df7, df8,\n",
    "        df9, df10, df11, df12,\n",
    "        df1, df2, df3, df4,\n",
    "       ]\n",
    "\n",
    "cols =  ['K D', 'K S', 'M D', 'M S']\n",
    "\n",
    "df_1 = pd.DataFrame(columns = cols)\n",
    "df_2 = pd.DataFrame(columns = cols)\n",
    "df_3 = pd.DataFrame(columns = cols)\n",
    "\n",
    "df_s = [df_1, df_2, df_3]\n",
    "\n",
    "for i in range(12):\n",
    "    num = i%4\n",
    "    df_s[i//4][cols[i%4]] = dfs[i][columns_order_3].loc['Type i interferon signaling pathway', :]\n",
    "    \n",
    "#Plot\n",
    "colors = sns.color_palette('Set2', 4)\n",
    "cm = 1/2.54\n",
    "f, axes = plt.subplots(1, 3, figsize =  (16*cm, 8*cm))\n",
    "\n",
    "names = ['IdentiPy\\nNSAF',\n",
    "         'MSFragger\\nNSAF',\n",
    "         'X!Tandem\\nNSAF',\n",
    "        ]\n",
    "\n",
    "\n",
    "for df_i, name, ax_i in list(zip(df_s, names, axes)):\n",
    "\n",
    "    df_i['K S'].plot(kind = 'bar', alpha = 0.9, position = 0, width = 0.18, color = colors[1], ax = ax_i)\n",
    "    df_i['K D'].plot(kind = 'bar', alpha = 0.9, position = 1, width = 0.18, color = colors[0], ax = ax_i)\n",
    "    df_i['M S'].plot(kind = 'bar', alpha = 0.9, position = 2, width = 0.18, color = colors[2], ax = ax_i)\n",
    "    df_i['M D'].plot(kind = 'bar', alpha = 0.9, position = 3, width = 0.18, color = colors[3], ax = ax_i)\n",
    "    \n",
    "    ax_i.tick_params(axis = 'x', labelsize = 7)\n",
    "    ax_i.set_xticks(ticks = [0, 1, 2, 3, 4, 5],\n",
    "                    labels = ['IFNg 48h', 'IFNg 24h', 'IFNg 4h', 'IFNa-2b 48h', 'IFNa-2b 24h', 'IFNa-2b 4h'])\n",
    "    ax_i.tick_params(axis = 'y', labelsize = 7)\n",
    "    ax_i.set_ylim(0, 45)\n",
    "    ax_i.set_xlim(-0.75, 5.5)\n",
    "    \n",
    "    ax_i.set_title(name, fontsize = 8)\n",
    "    \n",
    "    if name == 'IdentiPy\\nNSAF':\n",
    "        ax_i.legend(fontsize = 7, labels = ['kNN static', 'kNN dynamic', 'MinDet static', 'MinDet dynamic'],\n",
    "                   loc = 'upper left')\n",
    "        ax_i.set_ylabel('Type i interferon signaling pathway', fontsize = 7)  \n",
    "    else:\n",
    "        ax_i.set_yticklabels([])\n",
    "        \n",
    "plt.subplots_adjust(wspace = .05)\n",
    "f.savefig('/home/kae-13-1/Metrics/Fig3b.png', dpi = 600, bbox_inches = 'tight')\n",
    "f.savefig('/home/kae-13-1/Metrics/Fig3b.svg', dpi = 600, bbox_inches = 'tight')\n",
    "plt.close()    "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "13f4c5d7",
   "metadata": {},
   "source": [
    "# Fig 4"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fa7322b2",
   "metadata": {},
   "source": [
    "Input files contain calcuted metrics (step 6 from the pipeline). Metrics calculated for all the comparison are listed in one data frame (example can be found in the output)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "379642ab",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>virus test</th>\n",
       "      <th>MinDet_dyn</th>\n",
       "      <th>MinDet_semidyn</th>\n",
       "      <th>MinDet_stat</th>\n",
       "      <th>kNN_dyn</th>\n",
       "      <th>kNN_semidyn</th>\n",
       "      <th>kNN_stat</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>glioblastoma_2019_4114_UP</td>\n",
       "      <td>-inf</td>\n",
       "      <td>-0.084432</td>\n",
       "      <td>-1.869185</td>\n",
       "      <td>0.809599</td>\n",
       "      <td>12.868354</td>\n",
       "      <td>16.932845</td>\n",
       "      <td>17.245727</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>glioblastoma_2019_5522_UP</td>\n",
       "      <td>0.595494</td>\n",
       "      <td>34.687435</td>\n",
       "      <td>35.019501</td>\n",
       "      <td>36.310029</td>\n",
       "      <td>77.509072</td>\n",
       "      <td>77.509072</td>\n",
       "      <td>78.091990</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>glioblastoma_2019_3821_UP</td>\n",
       "      <td>0.489403</td>\n",
       "      <td>31.587172</td>\n",
       "      <td>31.587172</td>\n",
       "      <td>27.717461</td>\n",
       "      <td>40.373165</td>\n",
       "      <td>40.302790</td>\n",
       "      <td>40.223179</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>glioblastoma_2019_6067_UP</td>\n",
       "      <td>0.598332</td>\n",
       "      <td>41.746335</td>\n",
       "      <td>42.328077</td>\n",
       "      <td>43.508912</td>\n",
       "      <td>64.438614</td>\n",
       "      <td>64.066159</td>\n",
       "      <td>64.351277</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>a172_dbtrg_DBTRG_UP</td>\n",
       "      <td>1.010526</td>\n",
       "      <td>86.089896</td>\n",
       "      <td>94.003316</td>\n",
       "      <td>129.094433</td>\n",
       "      <td>123.821804</td>\n",
       "      <td>127.681505</td>\n",
       "      <td>130.584163</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                  Unnamed: 0  virus test  MinDet_dyn  MinDet_semidyn  \\\n",
       "0  glioblastoma_2019_4114_UP        -inf   -0.084432       -1.869185   \n",
       "1  glioblastoma_2019_5522_UP    0.595494   34.687435       35.019501   \n",
       "2  glioblastoma_2019_3821_UP    0.489403   31.587172       31.587172   \n",
       "3  glioblastoma_2019_6067_UP    0.598332   41.746335       42.328077   \n",
       "4        a172_dbtrg_DBTRG_UP    1.010526   86.089896       94.003316   \n",
       "\n",
       "   MinDet_stat     kNN_dyn  kNN_semidyn    kNN_stat  \n",
       "0     0.809599   12.868354    16.932845   17.245727  \n",
       "1    36.310029   77.509072    77.509072   78.091990  \n",
       "2    27.717461   40.373165    40.302790   40.223179  \n",
       "3    43.508912   64.438614    64.066159   64.351277  \n",
       "4   129.094433  123.821804   127.681505  130.584163  "
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "m_a = pd.read_csv('/home/kae-13-1/Metrics/metric_a_for_heatmap.csv')\n",
    "m_b1 = pd.read_csv('/home/kae-13-1/Metrics/metric_b1_for_heatmap.csv')\n",
    "m_b2 = pd.read_csv('/home/kae-13-1/Metrics/metric_b2_for_heatmap.csv')\n",
    "\n",
    "m_a = m_a.rename(columns = {'vir' : 'virus test'})\n",
    "m_b1 = m_b1.rename(columns = {'vir' : 'virus test'})\n",
    "m_b2 = m_b2.rename(columns = {'vir' : 'virus test'})\n",
    "\n",
    "cm = 1/2.54\n",
    "fig, axes = plt.subplots(3, 1, figsize = (8*cm, 22*cm))\n",
    "heatmap_df_data = [m_a, m_b1, m_b2]\n",
    "\n",
    "labels = list(m_a.columns[1:])\n",
    "labels = [' '.join(i.split('_')) for i in labels]\n",
    "labels = [i.replace('kNN', 'K') for i in labels]\n",
    "labels = [i.replace('MinDet', 'M') for i in labels]\n",
    "labels = [i.replace(' stat', ' S') for i in labels]\n",
    "labels = [i.replace(' dyn', ' D') for i in labels]\n",
    "labels = [i.replace('semidyn', 'SD') for i in labels]\n",
    "labels = [i.replace('virus test', 'V T') for i in labels]\n",
    "\n",
    "for df, ax in list(zip(heatmap_df_data, axes)):\n",
    "    mask = np.triu(np.ones_like(df.corr(), dtype = bool), k = 1)\n",
    "    f = sns.heatmap(df.corr(), vmin = 0.1, vmax = 1, annot = True, mask = mask, \n",
    "                    ax = ax, cmap = 'RdYlGn', fmt = '.3f', linewidth = .5, \n",
    "                    annot_kws = {'fontsize' : 6})\n",
    "    cbar = f.collections[0].colorbar\n",
    "    cbar.ax.tick_params(labelsize = 6)\n",
    "    f.set_xticklabels(labels = labels, rotation = 45, fontsize = 6)\n",
    "    f.set_yticklabels(labels = labels, rotation = 45, fontsize = 6)\n",
    "    i += 1  \n",
    "    \n",
    "plt.subplots_adjust(hspace = .15, bottom = .05, top = .99, left = .12, right = .95)\n",
    "fig.savefig('/home/kae-13-1/Metrics/Fig4.png', dpi = 600)\n",
    "fig.savefig('/home/kae-13-1/Metrics/Fig4.svg')\n",
    "plt.close()\n",
    "\n",
    "m_b2.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4210f977",
   "metadata": {},
   "source": [
    "# Fig 5"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6db56c73",
   "metadata": {},
   "source": [
    "Input file contains results of MTT testing"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "d697a9f7",
   "metadata": {},
   "outputs": [],
   "source": [
    "#### Fig 5a\n",
    "df = pd.read_csv('/home/kae-13-1/Metrics/mtt.csv')\n",
    "df['C'] = df['C'].astype(float)\n",
    "df['mean'] = df['mean'].astype(float)\n",
    "df['std'] = df['std'].astype(float)\n",
    "df['log'] = df['C'].apply(lambda x: np.log10(float(x)))\n",
    "\n",
    "colors = [sns.color_palette('RdYlGn', 5)[0]] + [sns.color_palette('RdYlGn', 5)[1]] + [sns.color_palette('RdYlGn', 5)[4]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "id": "418b7120",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>C</th>\n",
       "      <th>mean</th>\n",
       "      <th>std</th>\n",
       "      <th>type</th>\n",
       "      <th>log</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0.50000</td>\n",
       "      <td>61.323251</td>\n",
       "      <td>5.881612</td>\n",
       "      <td>24</td>\n",
       "      <td>-0.30103</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0.25000</td>\n",
       "      <td>59.319471</td>\n",
       "      <td>8.693449</td>\n",
       "      <td>24</td>\n",
       "      <td>-0.60206</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0.12500</td>\n",
       "      <td>87.939509</td>\n",
       "      <td>9.432601</td>\n",
       "      <td>24</td>\n",
       "      <td>-0.90309</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0.06250</td>\n",
       "      <td>80.907372</td>\n",
       "      <td>18.492745</td>\n",
       "      <td>24</td>\n",
       "      <td>-1.20412</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0.03125</td>\n",
       "      <td>84.272212</td>\n",
       "      <td>25.712343</td>\n",
       "      <td>24</td>\n",
       "      <td>-1.50515</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         C       mean        std  type      log\n",
       "0  0.50000  61.323251   5.881612    24 -0.30103\n",
       "1  0.25000  59.319471   8.693449    24 -0.60206\n",
       "2  0.12500  87.939509   9.432601    24 -0.90309\n",
       "3  0.06250  80.907372  18.492745    24 -1.20412\n",
       "4  0.03125  84.272212  25.712343    24 -1.50515"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "e6ea2c8e",
   "metadata": {},
   "outputs": [],
   "source": [
    "cm = 1/2.54\n",
    "def func(t, b, bline):\n",
    "    \n",
    "    def dose_response_graphpad(x, sl, x0):\n",
    "        fifty = (t + bline)/2\n",
    "        log_arg = (t - b)/(fifty - b) - 1\n",
    "        y = b + (t - b)/(1 + 10**(sl*(x0 - x) + np.log10(log_arg)))                                     \n",
    "        return y\n",
    "    \n",
    "    return dose_response_graphpad\n",
    "\n",
    "fig,  ax = plt.subplots(1, 1, figsize = (8*cm, 5*cm))\n",
    "\n",
    "j = 0\n",
    "for time_point, color in zip([24, 48, 72], colors):\n",
    "    \n",
    "    tmp = df[df['type'] == time_point]\n",
    "    xdata = tmp['log'].values\n",
    "    ydata = tmp['mean'].values\n",
    "    \n",
    "    top = max(ydata)\n",
    "    bottom = min(ydata)\n",
    "    bline = 5\n",
    "    \n",
    "    popt, pcov = curve_fit(func(top, bottom, bline), xdata, ydata, method='lm')\n",
    "    \n",
    "    y_pred = func(top, bottom, 0)(xdata, *popt)\n",
    "    \n",
    "    IC_50 = 10**(popt[1])\n",
    "    \n",
    "    dIC_50_log = np.sqrt(np.diag(pcov))[1]\n",
    "    dIC_50 = IC_50*np.log(10)*dIC_50_log\n",
    "    \n",
    "    txt_time_point = str(time_point) + 'h\\t'\n",
    "    txt_r2 = '$R^{2}$ ' + str(round(r2_score(ydata, y_pred), 2))\n",
    "    txt_ic = '\\tIC50 ' + str(round(IC_50,3)) + ' ± ' + str(round(dIC_50, 3)) \n",
    "    \n",
    "    txt =  txt_r2 + txt_ic\n",
    "\n",
    "    ax.text(0.3, 0.98 - j*0.1, txt, fontsize = 8, color = color,\n",
    "           horizontalalignment = 'left', verticalalignment = 'top', transform = ax.transAxes)\n",
    " \n",
    "    sns.scatterplot(x = 'log', y = 'mean', data =  tmp, color = color, s = 15, \n",
    "                    label = str(time_point) + 'h'\n",
    "                   )\n",
    "    x = np.linspace(-2.5, 0, 100)\n",
    "    y = func(top, bottom, 0)(x, *popt)\n",
    "    sns.lineplot(x = x, y = y, linestyle = '--', color = color, linewidth = 1)\n",
    "    j+=1\n",
    "\n",
    "# plt.title('Dose-response (X is Log-concentration: Absolute IC50)', fontsize = 14)\n",
    "plt.ylabel('Cell viability, %', fontsize = 8)\n",
    "plt.xlabel('$log_{10}(C)$', fontsize = 8)\n",
    "\n",
    "ax.tick_params(axis = 'x', which = 'major', labelsize = 7)\n",
    "ax.tick_params(axis = 'y', which = 'major', labelsize = 7)\n",
    "\n",
    "\n",
    "ax.text(-0.2, 1.1, '(a)', fontsize = 8,\n",
    "           horizontalalignment = 'left', verticalalignment = 'top', transform = ax.transAxes)\n",
    "\n",
    "plt.legend(loc = 'lower left', fontsize = 7)\n",
    "\n",
    "plt.savefig('/home/kae-13-1/Metrics/Fig5a.png', dpi = 600, bbox_inches = 'tight')\n",
    "plt.savefig('/home/kae-13-1/Metrics/Fig5a.svg', dpi = 600, bbox_inches = 'tight')\n",
    "\n",
    "plt.close()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d667bfc6",
   "metadata": {},
   "source": [
    "Input file contains calculated metrucs (step 6 from the pipeline)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "b77a4d42",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "IC50  =  0.0301 ± 0.008 \n"
     ]
    }
   ],
   "source": [
    "#### Fig 5b\n",
    "mtt_test = {'A2780 0.5' : 30.4584447929158,\n",
    "            'A2780 0.25' : 40.7158011253574,\n",
    "            'A2780 0.031' : 62.7063923992252,\n",
    "            'A2780 0.016' : 83.3871414076192,\n",
    "            'A2780 0.008' : 108.274144451619,\n",
    "            'A2780 0.004' : 101.171478645881}\n",
    "\n",
    "color_r2 = sns.color_palette('RdYlGn', 10)[8]\n",
    "colors = [sns.color_palette('RdYlGn', 5)[3]] + [sns.color_palette('RdYlGn', 5)[4]] + [sns.color_palette('RdYlGn', 5)[1]]\n",
    "\n",
    "files = ['/home/kae-13-1/Metrics/Nazarov_June2022/ChemProt_semi-dynamic_DE.csv']\n",
    "cm = 1/2.54      \n",
    "\n",
    "columns_order = ['A2780 DMSO', 'A2780 0.004', 'A2780 0.008', 'A2780 0.016', 'A2780 0.031', 'A2780 0.25', 'A2780 0.5']\n",
    "for f in  files:\n",
    "    fig, ax = plt.subplots(1, figsize = (8*cm, 5*cm))\n",
    "    df = pd.read_csv(f)\n",
    "    new_df = pd.DataFrame(index = columns_order)\n",
    "    new_df = pd.concat([new_df, df.set_index('dataname')[['b1', 'log10(b2)']]], axis = 1)\n",
    "    \n",
    "    new_df['concentration'] = new_df.index\n",
    "    new_df['concentration'] = new_df['concentration'].apply(lambda x: x.split('A2780 ')[1])\n",
    "    new_df['concentration'] = new_df['concentration'].apply(lambda x: x.replace('DMSO', '0.0001'))\n",
    "    new_df['concentration'] = new_df['concentration'].apply(lambda x: float(x))\n",
    "    new_df['new_C'] = new_df.concentration.apply(lambda x: np.log10(x))\n",
    "    \n",
    "    new_df = pd.concat([new_df, pd.DataFrame.from_dict(mtt_test, orient = 'index', columns = ['mtt_test'])], axis = 1)\n",
    "    \n",
    "    res = linregress(new_df['new_C'].values, new_df['b1'].values)\n",
    "    \n",
    "    new_df['linM1'] = new_df['new_C'].apply(lambda x: res.slope*x + res.intercept)\n",
    "    \n",
    "    tmp_1 = pd.concat([new_df[['b1', 'new_C']], new_df[['linM1','new_C']]], \n",
    "                      axis = 0)\n",
    "    tmp_1['val'] = tmp_1['b1'].fillna(0) + tmp_1['linM1'].fillna(0)\n",
    "    \n",
    "    sns.scatterplot(x = 'new_C', y = 'b1', data = new_df, color = colors[0], s = 15)\n",
    "    sns.lineplot(x = 'new_C', y = 'linM1', data = new_df, color = colors[0], linestyle =  'dotted')\n",
    "    sns.lineplot(x = 'new_C', y = 'val', data = tmp_1.reset_index(drop = True), \n",
    "                 color = colors[0], linestyle = 'None')\n",
    "\n",
    "    ax.set_xlabel('$log_{10}(C)$', fontsize = 8)\n",
    "    ax.set_ylabel('$\\pi_{1}$', fontsize = 8)\n",
    "    \n",
    "    ax.set_yticks(np.arange(0, 1501, 300))\n",
    "    ax.set_xticks(np.arange(-4, -0.49, 0.5))\n",
    "    \n",
    "    ax.tick_params(axis = 'x', which = 'major', labelsize = 7)\n",
    "    ax.tick_params(axis = 'y', which = 'major', labelsize = 7)\n",
    "    \n",
    "    text1 = '$R^{2} $ ' + str(round(res.rvalue**2, 3))\n",
    "    \n",
    "    ax.text(0.97, 0.8, text1, fontsize = 8, color = color_r2,\n",
    "               horizontalalignment = 'right', verticalalignment = 'top', transform = ax.transAxes)\n",
    "    \n",
    "    \n",
    "    IC50_conc_mtt = -1.3369219813673359 \n",
    "    \n",
    "    dIC50_conc_mtt = 0.06456106866764932\n",
    "    \n",
    "    ax.text(0.05, 0.95, 'IC50 0.046 ± 0.007 uM ', fontsize = 8, color = colors[1],\n",
    "           horizontalalignment = 'left', verticalalignment = 'top', transform = ax.transAxes)\n",
    "    \n",
    "    ax.axvline(IC50_conc_mtt, ymin = 0, color = colors[1], linewidth = 1, linestyle = '--')\n",
    "    \n",
    "    IC50_point = np.mean(new_df['b1'].values[1:])\n",
    "    dIC50_point = 0\n",
    "    \n",
    "    IC50_conc = (IC50_point - res.intercept)/res.slope\n",
    "    dIC50_conc = (res.slope*(res.intercept_stderr + dIC50_point) + (IC50_point - res.intercept) * res.stderr)/(res.slope)**2\n",
    "  \n",
    "    \n",
    "    ax.axvline(IC50_conc, ymin = 0, color = colors[2], linewidth = 1, linestyle = '--')\n",
    "    \n",
    "    IC50 = 10**(IC50_conc)\n",
    "    dIC50 = IC50*np.log(10)*dIC50_conc\n",
    "    \n",
    "    \n",
    "    ax.text(0.05, 0.85, '$\\pi_{1}50$ 0.030 ± 0.008 uM ', fontsize = 8, color = colors[2],\n",
    "           horizontalalignment = 'left', verticalalignment = 'top', transform = ax.transAxes)\n",
    "    \n",
    "    ax.text(-0.2, 1.1, '(b)', fontsize = 8,\n",
    "           horizontalalignment = 'left', verticalalignment = 'top', transform = ax.transAxes)\n",
    "    \n",
    "    \n",
    "    print('IC50  =  {} ± {} '.format(str(round(IC50, 4)), str(round(dIC50,3))))\n",
    "\n",
    "    plt.savefig('/home/kae-13-1/Metrics/Fig5b.png', dpi = 600, bbox_inches = 'tight')\n",
    "    plt.savefig('/home/kae-13-1/Metrics/Fig5b.svg', dpi = 600, bbox_inches = 'tight')\n",
    "\n",
    "    plt.close()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c5d857dc",
   "metadata": {},
   "source": [
    "Input files contains processes from GO terms enrichment analysis with calculated Enrichment and GO score (step 9 from the pipeline)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "85b80424",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0.0</th>\n",
       "      <th>0.004</th>\n",
       "      <th>0.008</th>\n",
       "      <th>0.016</th>\n",
       "      <th>0.031</th>\n",
       "      <th>0.25</th>\n",
       "      <th>0.5</th>\n",
       "      <th>Pearson</th>\n",
       "      <th>proccess</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>description</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Regulation of mitotic spindle assembly</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.611816</td>\n",
       "      <td>4.707353</td>\n",
       "      <td>0.996833</td>\n",
       "      <td>Regulation of mitotic spindle assembly</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DNA packaging</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>3.026266</td>\n",
       "      <td>5.325365</td>\n",
       "      <td>9.602180</td>\n",
       "      <td>0.966917</td>\n",
       "      <td>DNA packaging</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Nucleosome assembly</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>4.184400</td>\n",
       "      <td>5.904109</td>\n",
       "      <td>8.297159</td>\n",
       "      <td>0.905549</td>\n",
       "      <td>Nucleosome assembly</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DNA conformation change</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.48322</td>\n",
       "      <td>1.316662</td>\n",
       "      <td>8.405212</td>\n",
       "      <td>12.756819</td>\n",
       "      <td>16.334572</td>\n",
       "      <td>0.903094</td>\n",
       "      <td>DNA conformation change</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Nucleosome positioning</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>7.471313</td>\n",
       "      <td>8.032984</td>\n",
       "      <td>13.866524</td>\n",
       "      <td>0.894620</td>\n",
       "      <td>Nucleosome positioning</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        0.0  0.004    0.008     0.016  \\\n",
       "description                                                             \n",
       "Regulation of mitotic spindle assembly  0.0    0.0  0.00000  0.000000   \n",
       "DNA packaging                           0.0    0.0  0.00000  0.000000   \n",
       "Nucleosome assembly                     0.0    0.0  0.00000  0.000000   \n",
       "DNA conformation change                 0.0    0.0  1.48322  1.316662   \n",
       "Nucleosome positioning                  0.0    0.0  0.00000  0.000000   \n",
       "\n",
       "                                           0.031       0.25        0.5  \\\n",
       "description                                                              \n",
       "Regulation of mitotic spindle assembly  0.000000   2.611816   4.707353   \n",
       "DNA packaging                           3.026266   5.325365   9.602180   \n",
       "Nucleosome assembly                     4.184400   5.904109   8.297159   \n",
       "DNA conformation change                 8.405212  12.756819  16.334572   \n",
       "Nucleosome positioning                  7.471313   8.032984  13.866524   \n",
       "\n",
       "                                         Pearson  \\\n",
       "description                                        \n",
       "Regulation of mitotic spindle assembly  0.996833   \n",
       "DNA packaging                           0.966917   \n",
       "Nucleosome assembly                     0.905549   \n",
       "DNA conformation change                 0.903094   \n",
       "Nucleosome positioning                  0.894620   \n",
       "\n",
       "                                                                      proccess  \n",
       "description                                                                     \n",
       "Regulation of mitotic spindle assembly  Regulation of mitotic spindle assembly  \n",
       "DNA packaging                                                    DNA packaging  \n",
       "Nucleosome assembly                                        Nucleosome assembly  \n",
       "DNA conformation change                                DNA conformation change  \n",
       "Nucleosome positioning                                  Nucleosome positioning  "
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#### Fig 5c\n",
    "ffolder = '/home/kae-13-1/Metrics/Nazarov_June2022/GO'\n",
    "qprot = pd.DataFrame()\n",
    "for rf in listdir(ffolder):\n",
    "    if (rf.endswith('string.enrichment_updated.tsv')) and ('semi-dyn' in rf):\n",
    "        dif_out = path.join(ffolder,rf)\n",
    "        \n",
    "        d = pd.read_csv(dif_out, delimiter = '\\t')\n",
    "        \n",
    "        d.rename(columns = {'metric':'metric_'+rf.split('_GO')[0].split('A2780_')[1]}, inplace = True)\n",
    "\n",
    "        tmp = pd.DataFrame(list(zip(d['description'], d['metric_'+rf.split('_GO')[0].split('A2780_')[1]])), \n",
    "                           index = d['description'])\n",
    "        qprot = pd.concat([qprot, tmp], axis = 1)\n",
    "        qprot.drop(columns = [0], inplace = True)\n",
    "        qprot.rename(columns = {1:rf.split('_GO')[0].split('A2780_')[1]}, inplace = True)\n",
    "\n",
    "        qprot.rename(\n",
    "            columns = {'0.004':0.004, '0.5':0.5, '0.25':0.25, '0.016':0.016, '0.008':0.008,'K2': 1e-5, '0.031':0.031}, \n",
    "            inplace = True)\n",
    "\n",
    "qprot = qprot[[1e-5, 0.004, 0.008, 0.016, 0.031, 0.25, 0.5]]\n",
    "qprot = qprot.rename(columns = {1e-5: 0})\n",
    "qprot.fillna(0.0, inplace = True)\n",
    "qproT = qprot.T\n",
    "qproT['concentration'] = [i for i in [0, 0.004, 0.008, 0.016, 0.031, 0.25, 0.5]]\n",
    "\n",
    "r = 0.70\n",
    "t1 = qproT.corr(method ='pearson').concentration[abs(qproT.corr(method ='pearson').concentration) > r]\n",
    "t = qproT.corr(method ='pearson').concentration\n",
    "\n",
    "qprot = pd.concat([qprot, t], axis = 1)\n",
    "qprot['proccess'] = qprot.index\n",
    "\n",
    "## Select GO terms for which the changes in GO score is linearly correlated with the changes in concentration.\n",
    "metric = 3.5\n",
    "qprot.rename(\n",
    "            columns = {'concentration':'Pearson'}, \n",
    "            inplace = True)\n",
    "qprot = qprot.sort_values(by = ['Pearson'], ascending = False, axis = 0)\n",
    "\n",
    "qprot.to_csv('/home/kae-13-1/Metrics/GO.csv')\n",
    "mcp = qprot[(abs(qprot.Pearson) > r)&(qprot.max(axis=1) > metric)]\n",
    "mcp.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "e9e6fd2e",
   "metadata": {},
   "outputs": [],
   "source": [
    "cm = 1/2.54\n",
    "f, ax = plt.subplots(figsize = (10*cm,18*cm))\n",
    "sns.heatmap(mcp[[0, 0.004, 0.008, 0.016, 0.031, 0.25, 0.5, 'Pearson']], annot=True, center = 0,\n",
    "            cmap= 'RdYlGn', linewidths=2, linecolor='white',\n",
    "           fmt = '.2f', cbar = False, annot_kws = {'fontsize' : 7})\n",
    "ax.text(-0.99, 1.05, '(c)', fontsize = 8,\n",
    "           horizontalalignment = 'left', verticalalignment = 'top', transform = ax.transAxes)\n",
    "ax.text(0.5, 1.025,\n",
    "        'Pathway activity at diff. conc.', fontsize = 8,\n",
    "           horizontalalignment = 'center', verticalalignment = 'top', transform = ax.transAxes)\n",
    "plt.tick_params(axis = 'y', labelsize = 8)\n",
    "plt.tick_params(axis = 'x', labelsize = 8)\n",
    "plt.ylabel(None)\n",
    "plt.xlabel('Topotecan concentration, μM', fontsize = 8)\n",
    "plt.savefig('/home/kae-13-1/Metrics/Fig5c.png', dpi = 600, bbox_inches=\"tight\")\n",
    "plt.savefig('/home/kae-13-1/Metrics/Fig5c.svg', dpi = 600, bbox_inches=\"tight\")\n",
    "plt.close()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
